infrastructure/src/main/java/org/corfudb/infrastructure/orchestrator/actions/RestoreRedundancyMergeSegments.java

```diff
@@ -236,10 +320,9 @@ private void transferCommittedTail(CorfuRuntime runtime, Layout layout, long com
      * @param runtime A current runtime.
      * @return A retrieved trim mark.
      */
-    @VisibleForTesting
-    long trimLog(CorfuRuntime runtime) {
+    private long trimLog(CorfuRuntime runtime) throws TimeoutException {
 
-        long trimMark = runtime.getAddressSpaceView().getTrimMark().getSequence();
+        long trimMark = runtime.getAddressSpaceView().getTrimMark(false).getSequence();
```

Comment:
@PavelZaytsev  What I'm thinking of is we need to catch RuntimeException in the outermost catch block inperformStateTransfer's backoff retry, instead of catching individual TimeoutException | NetworkException | WrongEpochException | UnknownAddressInRangeException. If you have exceptions that you do not want to retry, catch them first and then RuntimeTimeException. Also one question is why we do not retry TransferSegmentException? This exception happens if a transfer batch fails because of Timeout, WrongEpoch, etc, why don't we retry the workflow in performStateTransfer?

Reply:
Ok I made it retry on RuntimeException except for IllegalStateException (when the node is unresponsive) and ServerNotReadyException (when the sequencer is not ready).

Reply:
Ok, fixed